dev-lang/rust/rust-1.74.1-r101.ebuild

```diff
@@ -648,13 +655,59 @@ mrustc_bootstrap() {
 	# These flags are used in every invocation of our bootstrap `cargo`.
 	local cargo_flags="--target ${CFG_COMPILER_HOST_TRIPLE} -j $(makeopts_jobs) --release --verbose"
 
+	# mrustc requires gcc, so disable libcxx
+	[[ "${LLVM_USE_LIBCXX}" == "1" ]] && unset LLVM_USE_LIBCXX
+
+	local llvm_config_wrapper_cxxflags=0
+
 	if use system-llvm; then
 		export LLVM_CONFIG="$(get_llvm_prefix)/bin/llvm-config"
+
+		# g++ doesn't have libc++'s header path built-in
+		local llvm_config_cxxflags=$(${LLVM_CONFIG} --cxxflags)
+		elog "Checking llvm-config --cxxflags: '${llvm_config_cxxflags}'"
+		[[ "${llvm_config_cxxflags}" =~ (^|[[:space:]])-stdlib=libc\+\+([[:space:]]|$) ]] && {
+			elog "Found LLVM CXXFLAGS has \"--stdlib=libc++\""
+			llvm_config_wrapper_cxxflags=1
+		}
 	else
 		llvm_bootstrap
 		export LLVM_CONFIG="${WORKDIR}/bootstrap/llvm/bin/llvm-config"
 	fi
 
+	elog "Using LLVM_CONFIG: ${LLVM_CONFIG}"
+
+	elog "Preparing wrapper of llvm-config"
+	cat > ${WORKDIR}/llvm-config <<-EOF || die
```

Comment:
1. This needs comments explaining what it's doing.
2. Please let's do it conditionally. Call tc-get-cxx-stdlib, check if it returned `libc++`. If it did, set a variable like `RUST_NEED_LIBCXX_WORKAROUND=1`, then later on, set the LLVM_CONFIG hacks only if that variable is `1`.